packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -436,18 +442,28 @@ export function getValidatorApi({
 
       const fork = config.getForkName(slot);
       // set some sensible opts
+      // builderSelection will be deprecated and will run in mode MaxProfit if builder is enabled
+      // and the actual selection will be determined using builderBoostFactor passed by the validator
       builderSelection = builderSelection ?? routes.validator.BuilderSelection.MaxProfit;
+      builderBoostFactor = builderBoostFactor ?? BigInt(100);
+      if (builderBoostFactor > MAX_BUILDER_BOOST_FACTOR) {
+        throw Error(`Invalid builderBoostFactor=${builderBoostFactor} > MAX_BUILDER_BOOST_FACTOR`);
+      }
+
       const isBuilderEnabled =
         ForkSeq[fork] >= ForkSeq.bellatrix &&
         chain.executionBuilder !== undefined &&
-        builderSelection !== routes.validator.BuilderSelection.ExecutionOnly;
+        builderSelection !== routes.validator.BuilderSelection.ExecutionOnly &&
+        builderBoostFactor !== BigInt(0);
```

Comment:
I think it is safer to change it to `builderBoostFactor > BigInt(0)` if for any reason we have a < 0 boost factor to avoid builder block being produced